### ISSUE TITLE
`calcite-input` clear button is not tab-able

### ISSUE THREAD

**Issue by minhiu** [Esri/calcite-design-system#11782]
### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

- In a "clearable" `calcite-input`, users cannot tab to the "clear" button.
- Is it because users can use `CTRL + A` to select all and delete? Would be nice to get feedbacks from Calcite team on why this "clear" button is not tab-able

### Actual Behavior

This "clear" button was given a `tab-index="-1"`

### Expected Behavior

This "clear" button should be tabbed to for keyboard users.

### Reproduction Sample

https://codepen.io/minhiu/pen/MYWGKdr?editors=100

### Reproduction Steps

1. Create a `calcite-input` with `clearable` prop
2. Observe that once a value is given to an input, users cannot tab to the "clear" button

### Reproduction Version

3.0.0

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Business/Community Analyst

**Comment by macandcheese** [comment]
This has come up before - in short - the current behavior is intentional and provides the best experience for keyboard only users.

Reference: https://github.com/Esri/calcite-design-system/issues/9718#issuecomment-2243602182

**Comment by minhiu** [comment]
Thanks a lot, @macandcheese, closing the issue now!
